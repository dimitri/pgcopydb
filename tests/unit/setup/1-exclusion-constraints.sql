DROP TABLE IF EXISTS exclcon;

CREATE TABLE exclcon (a integer, b integer);

insert into exclcon values (1,2), (3,4);

ALTER TABLE exclcon
  ADD CONSTRAINT constraint_fail_idx EXCLUDE USING btree (a WITH=);
