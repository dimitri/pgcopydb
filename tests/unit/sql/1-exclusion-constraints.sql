   select n.nspname, i.relname,
          rn.nspname, r.relname,
          indisprimary,
          indisunique,
          (select string_agg(attname, ',')
             from pg_attribute
            where attrelid = r.oid
              and array[attnum::integer] <@ indkey::integer[]
          ) as cols,
          pg_get_indexdef(indexrelid),
          c.conname,
          pg_get_constraintdef(c.oid)
     from pg_index x
          join pg_class i ON i.oid = x.indexrelid
          join pg_class r ON r.oid = x.indrelid
          join pg_namespace n ON n.oid = i.relnamespace
          join pg_namespace rn ON rn.oid = r.relnamespace
          left join pg_depend d
                 on d.classid = 'pg_class'::regclass
                and d.objid = i.oid
                and d.refclassid = 'pg_constraint'::regclass
                and d.deptype = 'i'
          left join pg_constraint c ON c.oid = d.refobjid
    where r.relkind = 'r' and r.relpersistence = 'p'
      and n.nspname !~ '^pg_' and n.nspname <> 'information_schema'
      and rn.nspname = 'public' and r.relname = 'exclcon'
 order by n.nspname, r.relname;
