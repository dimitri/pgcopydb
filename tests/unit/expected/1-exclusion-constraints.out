-[ RECORD 1 ]--------+-------------------------------------------------------------------
nspname              | public
relname              | constraint_fail_idx
nspname              | public
relname              | exclcon
indisprimary         | f
indisunique          | f
cols                 | a
pg_get_indexdef      | CREATE INDEX constraint_fail_idx ON public.exclcon USING btree (a)
conname              | constraint_fail_idx
pg_get_constraintdef | EXCLUDE USING btree (a WITH =)

